From 93946a315de320b5e15354e801863d85d6675520 Mon Sep 17 00:00:00 2001
From: Voon Weifeng <weifeng.voon@intel.com>
Date: Sat, 8 Jun 2019 20:40:37 +0800
Subject: [PATCH 50/52] REVERTME: net: stmmac: Temporary disable safety
 features

When the safety features is turn on, it will shows the below issue:
[  482.477702] stmmaceth 0000:00:1e.4 eth0: Found uncorrectable error in MAC: 'ATPES: Application Transmit Interface Parity Check Error'
[  482.489693] stmmaceth 0000:00:1e.4 eth0: Found uncorrectable error in MAC: 'RDPES: Read Descriptor Parity Check Error'
[  482.500365] stmmaceth 0000:00:1e.4 eth0: Found uncorrectable error in MAC: 'MPES: MTL Data Path Parity Check Error'
[  482.510776] stmmaceth 0000:00:1e.4 eth0: Found uncorrectable error in MAC: 'CWPES: CSR Write Data Path Parity Check Error'

Signed-off-by: Voon Weifeng <weifeng.voon@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_main.c | 3 ++-
 drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c  | 1 +
 include/linux/stmmac.h                            | 1 +
 3 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
index ce849e1..be8afcb 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -2577,7 +2577,8 @@ static int stmmac_hw_setup(struct net_device *dev, bool init_ptp)
 	stmmac_mtl_configuration(priv);
 
 	/* Initialize Safety Features */
-	stmmac_safety_feat_configuration(priv);
+	if (priv->plat->has_safety_feat)
+		stmmac_safety_feat_configuration(priv);
 
 	ret = stmmac_rx_ipc(priv, priv->hw);
 	if (!ret) {
diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
index 5b8d5a5..5fd532b 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
@@ -156,6 +156,7 @@ static int intel_mgbe_common_data(struct pci_dev *pdev,
 	plat->clk_trail_n = 2;
 	plat->has_gmac = 0;
 	plat->has_gmac4 = 1;
+	plat->has_safety_feat = 0;
 	plat->force_sf_dma_mode = 0;
 	plat->tso_en = 1;
 	plat->tsn_est_en = 1;
diff --git a/include/linux/stmmac.h b/include/linux/stmmac.h
index 512dc29..07202d8 100644
--- a/include/linux/stmmac.h
+++ b/include/linux/stmmac.h
@@ -199,5 +199,6 @@ struct plat_stmmacenet_data {
 	u32 ptov;
 	u32 ctov;
 	u32 tils;
+	bool has_safety_feat;
 };
 #endif
-- 
2.7.4

